README.md

```diff
@@ -1,2 +1,83 @@
-# puppetlabs-powershell_task_helper
-A helper library for Puppet Tasks written in PowerShell
+# powershell_task_helper
+
+A helper library for Puppet Tasks written in PowerShell.
+
+## Table of Contents
+
+1. [Description](#description)
+1. [Setup - The basics of getting started with powershell_task_helper](#setup)
+    * [Beginning with powershell_task_helper](#beginning-with-powershell_task_helper)
+1. [Usage - Configuration options and additional functionality](#usage)
+1. [Limitations - OS compatibility, etc.](#limitations)
+1. [Development - Guide for contributing to the module](#development)
+
+## Description
+
+A PowerShell helper library for use by [Puppet Tasks](https://puppet.com/docs/bolt/latest/writing_tasks.html). It provides a function that handles error generation, simplifies JSON input and output, and makes testing your task easier. It requires Bolt >= 2.4.
+
+## Setup
+
+### Beginning with powershell_task_helper
+
+To use this library, include this module in a `Puppetfile`
+
+```ruby
+mod 'puppetlabs-powershell_task_helper'
+```
+
+Add it to your [task metadata](https://puppet.com/docs/bolt/1.x/writing_tasks.html#concept-677)
+
+```json
+{
+  "files": ["powershell_task_helper/files/BoltPwshHelper/"],
+  "input_method": "powershell"
+}
+```
+
+## Usage
+
+Bolt automatically adds your task directory to the `$env:PSModulePath`, so PowerShell module auto loading will work without having to write out the `Import-Module` statement. If your target is using PowerShell 3.0, you will have to add an import statement.
+
+Below is a short example. For a more complete example look at `examples/mytask.ps1` in this project.
+
+```powershell
+#!/usr/bin/env pwsh
+[CmdletBinding()]
+Param(
+  [Parameter(Mandatory = $True)]
+  [String]
+  $Name
+)
+
+<#
+If using PowerShell 3.0, you will need to add an import statement here:
+Import-Module BoltPwshHelper
+#>
+​
+trap {
+  Write-BoltError -Message "Generic trap handler" -Exception $_
+}
+```
+
+You can then run the task like any other Bolt task
+
+```shell
+bolt task run mymodule::task --targets target.example.com name='Robert'
+```
+
+Or using the Bolt PowerShell cmdlets:
+
+```shell
+Invoke-BoltTask mymodule::task -t target.example.com name='Robert'
+```
+
+
+## Limitations
+
+## Development
+
+Puppet modules on the Puppet Forge are open projects, and community contributions are essential for keeping them great. We can’t access the huge number of platforms and myriad of hardware, software, and deployment configurations that Puppet is intended to serve.
+
+We want to keep it as easy as possible to contribute changes so that our modules work in your environment. There are a few guidelines that we need contributors to follow so that we can have a chance of keeping on top of things.
+
+For more information, see the CONTRIBUTING.md documentation in this module.
```

Comment:
This repo doesn't have a contributing doc currently. We can reuse [this one](https://github.com/puppetlabs/puppetlabs-ruby_task_helper/blob/main/CONTRIBUTING.md) and remove the section on testing since this repo doesn't have any tests.

Reply:
Added your example contributing file